### DISCUSSION TITLE
Anyway to run .apk files on school chromebook?

### DISCUSSION THREAD

**Discussion by CALR3F** [3kh0/ext-remover#514] [Help JOIN TITANIUM NETWORK TO SHARE PLEASE] [answered]
Google play along with almost everything on the webstore is blocked.

**Comment by fnfgo** [comment]
Arc welder used to work but it got removed from the webstore just use sh1mmer

**Comment by obkduishinijf** [comment]
just use sh1mmer 🤷

**Reply by fnfgo** [reply]
> just use sh1mmer 🤷

ON GOD STOP COPYING WHAT I SAID

**Reply by obkduishinijf** [reply]
> > just use sh1mmer 🤷
> 
> ON GOD STOP COPYING WHAT I SAID

it's not copying, who the fuck copies words? it's just us both stating the same thing.

**Reply by xHydra1337** [reply]
> > just use sh1mmer 🤷
> 
> ON GOD STOP COPYING WHAT I SAID

Bro hate to say it but your kinda being stingy here, the only way to use apk's is to just unenroll, we all have the same idea here, its not copying. Copying would be stating exactly what you said work for word.

**Reply by obkduishinijf** [reply]
> > > just use sh1mmer 🤷
> > 
> > 
> > ON GOD STOP COPYING WHAT I SAID
> 
> Bro hate to say it but your kinda being stingy here, the only way to use apk's is to just unenroll, we all have the same idea here, its not copying. Copying would be stating exactly what you said work for word.

fr

**Reply by fnfgo** [reply]
> > > > just use sh1mmer 🤷
> > > 
> > > 
> > > ON GOD STOP COPYING WHAT I SAID
> > 
> > 
> > Bro hate to say it but your kinda being stingy here, the only way to use apk's is to just unenroll, we all have the same idea here, its not copying. Copying would be stating exactly what you said work for word.
> 
> fr

Arc welder used to work but it got removed from the webstore just use sh1mmer
youre actually blind dawg he did say the same thing but with an emoji after it agian i said this:just use sh1mmer he said this:just use sh1mmer 🤷

**Reply by v4pl** [reply]
sh1mmer got patched and i'm on fucking 121

**Reply by obkduishinijf** [reply]
p sure the site has an explanation for how to do it if it's patched, involves taking the back panel and battery out though

**Reply by kosuke14** [reply]
you can exploit arcvm on v112 without unenrollment, dev mode, and arc welder if you have access to android.
you can downgrade to v112 from v121.

**Reply by BroJac5246** [reply]
> you can exploit arcvm on v112 without unenrollment, dev mode, and arc welder if you have access to android.

Can you explain how to do that? Also, will it work on v114 or is v112 the last version to support it?

**Reply by kosuke14** [reply]
It won't work on v114.
v113 and older will work.

Downgrading ChromeOS will need a computer which you have access and USB memory.
Installing .apk files will need a computer that has installed ADB, and Wi-Fi. ofc you need access to both them

Recovery Image could be found at https://chromiumdash.appspot.com/serving-builds?deviceCategory=ChromeOS if you can't find the version you want, try using https://web.archive.org (you need 113 or 112 on your board)
Use Extension called "Chrome Recovery Utility" to write the Image.
Downgrading Steps will be found at ext-remover's README. Follow them.
https://github.com/3kh0/ext-remover?tab=readme-ov-file#downgrading
Note that it will delete all local data in your Chromebook, it doesn't mean unenrollment.

When you get version 113 or 112 successfully, go to settings and find the android settings.
Then, scroll down and find Version Info, and hit Build Version for about 20 times.
Once Developer Options in Android is enabled, go back to android settings home and go to search bar (press the search icon button top of the screen)
and type "pair" and click the result (it will appear 2 options)
Check you're connected both computer and Chromebook to the same Wi-Fi and enable "Wireless Debugging".
Press 'Trust' on your Chromebook.
Connect to your Chromebook with your computer by using `adb connect <IP ADDRESS>` command.
Press 'Trust' on your Chromebook.
Get your apk on your computer and you can install them by using `adb install <path to apk>` command.
I recommend to install https://www.apkmirror.com/apk/google-inc/package-installer/package-installer-11-6557492-release/package-installer-11-6557492-android-apk-download/ it can bypass .apk install restriction.

**Reply by BroJac5246** [reply]
I've heard about that before but thanks for giving more details!

> hit Build Version for about 20 times

BTW it's 7 times

**Comment by BroJac5246** [comment, marked as answer]
Not unless:
A. You have developer mode on
B. Your school was somehow stupid enough to let you enable Linux. If that somehow is the case, you could enable ADB and continue from there. However, there would be a warning at the bottom of the lock screen that won't go away until a powerwash.

There also might be some obscure exploit, but it probably wouldn't be worth it.

**Comment by kosuke14** [comment]
If your chromebook has Android 11+ and Chrome V108 and lower, you could enable "wireless debugging" on android developer settings to install apk from outside but it is detectable the installations if your chromebook is managed.
Also, you could install older version of Package Installer to install your apk in the future.

**Reply by BroJac5246** [reply]
> If your chromebook has Android 11+ and Chrome V108 and lower, you could enable "wireless debugging" on android developer settings to install apk from outside but it is detectable the installations if your chromebook is managed.

I'm not able to do this— my district has blocked enabling Android Developer Settings. Were you able to successfully perform this on your managed Chromebook?

**Reply by kosuke14** [reply]
Yes, I've successfully performed that on my managed Chromebook before updating. (on ChromeOS v108)
I've tried these steps.
1. Open Settings and go to "Google Play Store" > "Android Settings"
2. Scroll down and find "System Informations"
3. Smash the "Build Version"
4. Go back to android settings home and find "Developer Options"
5. Find "Wireless debugging" and turn on.
6. Launch shell on another device and connected with adb by using ip address.
7. Perform "adb install <.apk>"

**Reply by AshtonDavies** [reply]
appetize.io
Runs APK files in JavaScript in the browser now apparently.

**Reply by BroJac5246** [reply]
> appetize.io
> Runs APK files in JavaScript in the browser now apparently.

That just streams from an Android emulator (I think it's the one included in Android Studio)

**Reply by BroJac5246** [reply]
> Yes, I've successfully performed that on my managed Chromebook before updating.

How does that not require ChromeOS Developer Mode!!! I thought ADB is supposed to be completely off-limits if you don't have ChromeOS Developer Mode or that Linux ADB setting, but I guess not... I would test but I've enabled ADB via Linux on my personal Chromebook so I can't confirm this.

**Reply by kosuke14** [reply]
The way I've tried doesn't require dev mode and linux, it only requires your own wifi, adb installed computer and your chromebook with ChromeOS v108 and lower.

**Reply by BroJac5246** [reply]
> your chromebook with ChromeOS v108 and lower

Have you confirmed this doesn't work on newer versions?

**Reply by kosuke14** [reply]
> Have you confirmed this doesn't work on newer versions?

Yes, I've comfirmed that "Wireless debugging" option was grayed out on ChromeOS v110 and newer. but v109 is not comfirmed.

**Comment by kosuke14** [comment]
I've confirmed that it's able to install apk files in **ChromeOS v112 and lower**.
ChromeOS v112 is **able to downgrade** from v121 for me.
**Without developer mode nor Linux**
First, go to Android Settings and enable developer options on **Android**.
Second, *search* for 'paired' on Android Settings.
Now, I can enable 'Wireless Debugging'.
Connect ADB with Chromebook and able to install apk.
This method has been patched on ChromeOS v114 and higher.

**Reply by kosuke14** [reply]
To downgrade,
ChromeOS v112 Recovery Image could be found at https://web.archive.org/web/20230816060506/https://chromiumdash.appspot.com/serving-builds?deviceCategory=ChromeOS

chrome100.dev has only legacy version, you may be not able to find v112 image at there.

**Reply by BroJac5246** [reply]
> This could be fixed on ChromeOS v114 and higher.

Doesn't work for me on v114.

v120 should be rolling out to the LTS channel in about a month so I'm not sure I would want to be on v112, assuming I'd even be able to downgrade to it.

**Reply by kosuke14** [reply]
This method has been patched on ChromeOS v114 (or v113?) and higher.

I was able to downgrade to v112 from v121 by using official recovery image without unenrollment.

**Reply by kosuke14** [reply]
This bug is named CVE-2023-3742
It was patched on v114.

**Comment by ilovetoucans** [comment]
go into developer mode

**Reply by BroJac5246** [reply]
That's already been suggested.
